File: benchmark/objectivec/task5_class.m
#import <Foundation/Foundation.h>
#import <math.h>

@interface Point : NSObject
@property double x;
@property double y;
- (instancetype)initWithX:(double)x y:(double)y;
- (double)distanceTo:(Point*)other;
@end

@implementation Point
- (instancetype)initWithX:(double)x y:(double)y {
    self = [super init];
    if (self) {
        _x = x;
        _y = y;
    }
    return self;
}

- (double)distanceTo:(Point*)other {
    double dx = self.x - other.x;
    double dy = self.y - other.y;
    return sqrt(dx * dx + dy * dy);
}
@end

File: benchmark/prolog/task5_class.pl
point(X, Y, point(X, Y)).

distance_to(point(X1, Y1), point(X2, Y2), D) :-
    DX is X1 - X2,
    DY is Y1 - Y2,
    D is sqrt(DX * DX + DY * DY).
